src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/PublicDataRetrievalService.java

```diff
@@ -0,0 +1,76 @@
+package uk.gov.companieshouse.overseasentitiesapi.service;
+
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.handler.exception.URIValidationException;
+import uk.gov.companieshouse.api.model.ApiResponse;
+import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
+import uk.gov.companieshouse.api.model.officers.OfficersApi;
+import uk.gov.companieshouse.overseasentitiesapi.client.ApiClientService;
+import uk.gov.companieshouse.overseasentitiesapi.exception.ServiceException;
+import uk.gov.companieshouse.overseasentitiesapi.utils.ApiLogger;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.HashMap;
+
+@Component
+public class PublicDataRetrievalService {
+    public static final String COMPANY_NUMBER = "company_number";
+    public static final String OFFICER_NAME = "officer_name";
+
+    private final ApiClientService apiClientService;
+
+    public PublicDataRetrievalService(ApiClientService apiClientService) {
+        this.apiClientService = apiClientService;
+    }
+
+    public void initialisePublicData(String companyNumber, String passThroughTokenHeader) throws ServiceException {
+        getCompanyProfile(companyNumber, passThroughTokenHeader);
```

Comment:
Please store the returns in two public fields (private member with getter/setter)
Also info log to show the date fetched

Reply:
ok log done below